ansi/halfblocks/encoder.go

```diff
@@ -0,0 +1,626 @@
+package halfblocks
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+	"math"
+	"strings"
+)
+
+// Block represents different Unicode block characters
+type Block struct {
+	Char        rune
+	Coverage    []bool // Which parts of the block are filled (true = filled)
+	CoverageMap string // Visual representation of coverage for debugging
+}
+
+// EncoderOptions contains all configurable settings
+type EncoderOptions struct {
+	ColorMode    int     // 0=none, 1=8colors, 2=256colors, 3=truecolor
```

Comment:
we probably don't need this since downsampling colors is handled by `colorprofile`